### PR TITLE
add copy image

### PR DESCRIPTION
**Pull request by 956081712**
add resource copy image

### CODE REVIEW

File: alicloud/resource_alicloud_copy_images.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+			imageTags = append(imageTags, imageTag)
+		}
+		request.Tag = &imageTags
+	}
+
+	raw, err := client.WithEcsClient(func(ecsClient *ecs.Client) (interface{}, error) {
+		return ecsClient.CopyImage(request)
+	})
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_image", request.GetActionName(), AlibabaCloudSdkGoERROR)
```
Comment (Lexsss) [inline alicloud/resource_alicloud_copy_images.go]: alicloud_copy_image

File: alicloud/resource_alicloud_copy_images.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		if NotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return WrapError(err)
+	}
+
+	d.Set("name", object.ImageName)
+	d.Set("description", object.Description)
+
```
Comment (Lexsss) [inline alicloud/resource_alicloud_copy_images.go]: maybe set more fields

File: alicloud/resource_alicloud_copy_images.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+	stateConf := BuildStateConf([]string{"Creating"}, []string{"Available"}, d.Timeout(schema.TimeoutCreate), 1*time.Minute, ecsService.ImageStateRefreshFunc(d.Id(), []string{"CreateFailed", "UnAvailable"}))
+	if _, err := stateConf.WaitForState(); err != nil {
+		return WrapErrorf(err, IdMsg, d.Id())
+	}
+	return resourceAliCloudCopyImageRead(d, meta)
+}
+func resourceAliCloudCopyImageUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	ecsService := EcsService{client}
+	err := ecsService.updateImage(d)
```
Comment (Lexsss) [inline alicloud/resource_alicloud_copy_images.go]: call api in this function please

File: alicloud/resource_alicloud_copy_images.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	tags := object.Tags.Tag
+	if len(tags) > 0 {
+		err = d.Set("tags", tagsToMap(tags))
+	}
+	return WrapError(err)
+}
+
+func resourceAliCloudCopyImageDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	ecsService := EcsService{client}
```
Comment (Lexsss) [inline alicloud/resource_alicloud_copy_images.go]: call api in this function please

File: alicloud/resource_alicloud_copy_images_test.go [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckImageExistsWithProviders(resourceId, &v, &providers),
+					testAccCheck(map[string]string{
+						"name": fmt.Sprintf("tf-testAccEcsImageConfigBasic%dchange", rand),
+					}),
+				),
+			},
+			{
+				Config: testAccConfig(map[string]interface{}{
+					"tags": map[string]string{
```
Comment (Lexsss) [inline alicloud/resource_alicloud_copy_images_test.go]: add capital letter check to tags please

File: alicloud/resource_alicloud_copy_images_test.go [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+  region = "cn-hangzhou"
+}
+data "alicloud_instance_types" "default" {
+    provider = "alicloud.hz"
+ 	cpu_core_count    = 1
+	memory_size       = 2
+}
+data "alicloud_images" "default" {
+  provider = "alicloud.hz"
+  name_regex  = "^ubuntu*"
```
Comment (Lexsss) [inline alicloud/resource_alicloud_copy_images_test.go]: "^ubuntu_18.*_64"

File: alicloud/resource_alicloud_image.go [hunk 7]
```diff
@@ -206,46 +205,19 @@ func resourceAliCloudImageCreate(d *schema.ResourceData, meta interface{}) error
...
 	}
 	return resourceAliCloudImageRead(d, meta)
 }
 func resourceAliCloudImageUpdate(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*connectivity.AliyunClient)
-	d.Partial(true)
-
-	err := setTags(client, TagResourceImage, d)
+	ecsService := EcsService{client}
+	err := ecsService.updateImage(d)
```
Comment (Lexsss) [inline alicloud/resource_alicloud_image.go]: still

File: alicloud/resource_alicloud_image.go [hunk 8]
```diff
@@ -276,38 +248,8 @@ func resourceAliCloudImageRead(d *schema.ResourceData, meta interface{}) error {
...
-		return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
-	}
-	addDebug(request.GetActionName(), raw)
-	stateConf := BuildStateConf([]string{}, []string{}, d.Timeout(schema.TimeoutCreate), 3*time.Second, ecsService.ImageStateRefreshFunc(d.Id(), []string{}))
-	if _, err := stateConf.WaitForState(); err != nil {
-		return WrapErrorf(err, IdMsg, d.Id())
-	}
-
-	return nil
+	return ecsService.deleteImage(d)
```
Comment (Lexsss) [inline alicloud/resource_alicloud_image.go]: still

File: alicloud/service_alicloud_ecs.go [hunk 9]
```diff
@@ -579,6 +579,77 @@ func (s *EcsService) DescribeImageById(id string) (image ecs.Image, err error) {
...
+	request.ImageId = object.ImageId
+
+	raw, err := s.client.WithEcsClient(func(ecsClient *ecs.Client) (interface{}, error) {
+		return ecsClient.DeleteImage(request)
+	})
+
+	if err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	addDebug(request.GetActionName(), raw)
```
Comment (Lexsss) [inline alicloud/service_alicloud_ecs.go]: complete addDebug and move this api call to resource.go please

File: alicloud/resource_alicloud_copy_images.go [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"image_id": {
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_copy_images.go]: please use `source_image_id` instead of `image_id`

**Pull request merged**